src/output-json-alert.c

```diff
@@ -93,12 +93,13 @@ typedef struct JsonAlertLogThread_ {
 
 /* Callback function to pack payload contents from a stream into a buffer
  * so we can report them in JSON output. */
-static int AlertJsonPrintStreamSegmentCallback(const Packet *p, void *data, uint8_t *buf, uint32_t buflen)
+static int AlertJsonDumpStreamSegmentCallback(const Packet *p, void *data, uint8_t *buf, uint32_t buflen)
 {
     MemBuffer *payload = (MemBuffer *)data;
 
-    PrintStringsToBuffer(payload->buffer, &payload->offset, payload->size,
-                         buf, buflen);
+    // We can't user Print...ToBuffer functions as we need real raw data.
+    memcpy(payload->buffer + payload->offset, buf, buflen);
```

Comment:
It looks like the PrintStringsToBuffer function that is replaced here does check the size of payload->buffer (quite indirectly in the PrintBufferData macro). Is it safe to unconditionally (and without boundary checking) copy data to payload->buffer here?